evalml/pipelines/component_graph.py

```diff
@@ -54,7 +54,10 @@ def from_list(cls, component_list, random_seed=0):
 
             component_dict[component_name] = [component_class]
             if previous_component is not None:
-                component_dict[component_name].append(f"{previous_component}.x")
+                if "sampler" in previous_component:
```

Comment:
@bchen1116 This assumes the undersampler is placed right before the final estimator. It makes sense for the sampling to happen there, and I know the pipelines that automl will create will follow that convention but I wonder what should happen when a user defines a pipeline where that isn't the case. 

I don't think it's fruitful for us to catch all possible "bad states" the user could find themselves in but I do think it's easier to find yourself in a bad state with the list representation. 

I don't think any action is required here but I wonder what other people think! Maybe it merits a broader discussion: @chukarsten @dsherry @angela97lin @jeremyliweishih @ParthivNaresh

Reply:
Ahh, this is a good point. I believe this code technically allows a pipeline to be initialized with a component graph similar to `['Undersampler', 'Imputer', 'Logistic Regression Classifier']`, although since we only track `y` for the undersampler transform output, we would run into issues during the `fit` call. 

This works:
![image](https://user-images.githubusercontent.com/22552445/112682457-83802380-8e46-11eb-9bb5-bb77e09b92aa.png)

This fails at the estimator:
![image](https://user-images.githubusercontent.com/22552445/112682481-8aa73180-8e46-11eb-9210-0063d49f1589.png)
![image](https://user-images.githubusercontent.com/22552445/112682551-a0b4f200-8e46-11eb-9d61-da77942f3951.png)

My goal was to make the sampler precede the final estimator, but it could be smart to ensure the sampler(s) come directly before the final estimator. It's certainly something we can validate through the `ComponentGraph` during instantiation. What do ya'll think?

Reply:
After stand-up, we decided to proceed with this approach and allow this to error, although we file an [issue](https://github.com/alteryx/evalml/issues/2053) to track adding support for tracking the `y` output of components when we initialize a `componentGraph` through a list.